Mathlib/RingTheory/Polynomial/Basic.lean

```diff
@@ -641,6 +698,23 @@ theorem _root_.Polynomial.coeff_prod_mem_ideal_pow_tsub {ι : Type*} (s : Finset
 
 end CommSemiring
 
+section Semiring
+
+/-- If `R` is a nontrivial ring, the polynomials `R[X]` are not finite as an `R`-module. When `R` is
+a field, this is equivalent to `R[X]` being an infinite-dimensional vector space over `R`.  -/
+theorem polynomial_not_module_finite {R : Type u} [Semiring R] [Nontrivial R]:
```

Comment:
I wonder if it's possible to deduce this from [Polynomial.basisMonomials](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/MvPolynomial/Basic.html#Polynomial.basisMonomials), but that currently only works for `CommSemiringR`, and I don't know how difficult it would be to generalize. So no need to do anything in this PR.

Reply:
I've also noticed that if you allow for stronger assumptions like `R` being a field, you can pretty swiftly get that `R[X]` has countable dimension by using the work from the MvPolynomial module, e.g. [rank_mvPolynomial](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/MvPolynomial.html#MvPolynomial.rank_mvPolynomial). But without at least `CommSemiring R`, I agree not much more can be done.